releasenotes/notes/edge-index-map-cf07a035d02481a1.yaml

```diff
@@ -0,0 +1,14 @@
+---
+features:
+  - |
+    Added a new method, :meth:`~retworkx.PyDiGraph.edge_index_map`, to the
+    :class:`~retworkx.PyDiGraph` and :class:`~retworkx.PyGraph`
+    (:meth:`~retworkx.PyGraph.edge_index_map`) that will return a read-only
+    mapping of edge indices to a tuple of the form
+    ``(node index, node_index, weight/data payload)`` for every edge in the
+    graph object.
+  - |
+    Added a new custom return type :class:`~retworkx.EdgeIndexMap` which is
+    returned by :class:`retworkx.PyDiGraph.edge_index_map` and
+    :class:`retworkx.PyGraph.edge_index_map`. It is equivalent to a read-only
+    dict/mapping that represent a mapping of edge indices to the edge.
```

Comment:
so is the correct way to identify an edge in retworkx to use the tuple (node1, node2, weight)? There's no Edge class or something?

Reply:
Yeah more or less, the unique identifier for an edge is it's id/index. But other than that it's primarily represented by the tuple. There is not a dedicated class for an edge or node (mostly because we let the data payload for either be any python object).

Reply:
Although I did intend for this sentence to be `edge tuple` instead of `edge` (I just made a typo).